### PR TITLE
A few functions I needed for my own work.

### PR DESCRIPTION
**Pull request by jeffreyrosenbluth**

### CODE REVIEW

File: Codec/Picture/Extra.hs [hunk 1]
```diff
@@ -130,3 +133,38 @@ rotateRight90 img@Image {..} =
   where
     gen x y = pixelAt img y (imageHeight - 1 - x)
 {-# INLINEABLE rotateRight90 #-}
+
+
+-- | Rotate image by 180, i.e flip both vertically and horizontally.
+
+rotate180 :: Pixel a => Image a -> Image a
```
Comment (mrkkrp) [inline Codec/Picture/Extra.hs]: If we are to add rotation by 180°, it makes sense to have two versions: rotate to the left and rotate to the right, as they apparently produce not the same result.

File: Codec/Picture/Extra.hs [hunk 2]
```diff
@@ -130,3 +133,38 @@ rotateRight90 img@Image {..} =
...
+rotate180 img@(Image w h _) = generateImage g w h
+  where
+    g x y = pixelAt img (w - 1 - x) (h - 1 - y)
+{-# INLINEABLE rotate180 #-}
+
+-- | Create an image by placing two images side by side.
+--   If the images are of differnet heights the smaller height is used.
+
+beside :: Pixel a => Image a -> Image a -> Image a
```
Comment (mrkkrp) [inline Codec/Picture/Extra.hs]: Would it be better to provide a function to "concatenate" arbitrary number of images in this way? Seems to more useful.

File: Codec/Picture/Extra.hs [hunk 3]
```diff
@@ -130,3 +133,38 @@ rotateRight90 img@Image {..} =
...
+      | x < w1 = pixelAt img1 x
+      | otherwise = pixelAt img2 (x - w1)
+    h = min h1 h2
+{-# INLINEABLE beside #-}
+
+-- | Create an image by placing the first iamge on top of the second
+--   If the images are of differnet widths the smaller width is used.
+
+below :: Pixel a => Image a -> Image a -> Image a
+below img1@(Image w1 h1 _) img2@(Image w2 h2 _) =
```
Comment (mrkkrp) [inline Codec/Picture/Extra.hs]: The same here.

**Comment by mrkkrp** [comment]
The error messages are strange, but CI still succeeds on master. Let's restart it for this PR.

**Comment by mrkkrp** [comment]
Have you by any change changed permissions of test pictures and committed that? It looks like something of this sort, because CI can't open the test images, and I see commited images which are identical to those I had, at least visually.

File: Codec/Picture/Extra.hs (hunk 1 above)
Comment (jeffreyrosenbluth) [inline Codec/Picture/Extra.hs]: I'm a bit puzzled here, can you explain why they don't?

File: Codec/Picture/Extra.hs (hunk 2 above)
Comment (jeffreyrosenbluth) [inline Codec/Picture/Extra.hs]: Sure, I only had use for the two image version but I'll generalize when I get a few minutes.

**Comment by jeffreyrosenbluth** [comment]
I didn't make any changes to the permissions, I'm puzzled by the CI failure as well.

File: Codec/Picture/Extra.hs (hunk 1 above)
Comment (mrkkrp) [inline Codec/Picture/Extra.hs]: Sorry, my fault, for some reason I thought they should differ, but now I see that they in fact do not. Please ignore this comment.

**Comment by mrkkrp** [comment]
To check, please amend your commit so we don't have a diff for this file: https://github.com/mrkkrp/JuicyPixels-extra/pull/5/files#diff-2f6cd55cf062773cdf3ce12a48ed4e2b, because it should not change. With that change, at least old tests should start to pass I think.

**Comment by jeffreyrosenbluth** [comment]
I think the test are failing because for some reason JuicyPixels things the example photos are of pixel type RGBA not RGB. I can fix this if you like.

**Comment by mrkkrp** [comment]
Hmm, I'll take a look myself when I have time. Ping me when you think the featrues are complete, I'll play with the PR locally and apply it if I manage to fix the tests (I should).

**Comment by jeffreyrosenbluth** [comment]
I think that does ti for now. There are many more functions that I can think of and perhaps one day when have more time I'll add them.

**Comment by mrkkrp** [comment]
Great, I'll take it from here, fix the tests (probbaly this weekend), and release a new version. Thanks for your contribution!

**Comment by jeffreyrosenbluth** [comment]
Thanks

**Comment by mrkkrp** [comment]
Applied as c713a3d68b4924242b811adfeec87b37cde39b53, I'll release version 0.2.0 now.

**Pull request closed without merging**